pkg/pillar/netdump/tar.go

```diff
@@ -0,0 +1,94 @@
+package netdump
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/user"
+	"strings"
+	"time"
+)
+
+type fileForTar struct {
+	dstPath string
+	info    fs.FileInfo
+	isDir   bool
+	content io.Reader
+}
+
+// Create .tar.gz archive at the location <tarPath> containing <files> inside.
+func createTarGz(tarPath string, files []fileForTar) (err error) {
```

Comment:
Is there any max file size of file which we will tar?

Reply:
Currently there isn't, but it is bound to some reasonable max size indirectly. The snapshot of the EVE and linux config varies in size minimally and it is unlikely going to contribute more than with few kilobytes (device config would have to be really large for this to make noticeable contribution to the tar file size). The recorded network traces are more dynamic in size. For zedcloud requests they are quite small and for downloader we disable some things (e.g. socket trace) to make sure that even if large image is being downloaded, the recorded network trace has only few KBs. Finally, packet captures (included only with /ping requests) are limited in size (by default to 1MB).